src/bin/signal_handler.rs

```diff
@@ -62,23 +74,49 @@ mod imp {
                         info!("{}", v)
                     }
                 }
-                SIGUSR2 => {
-                    if backup_path.is_empty() {
-                        info!("empty backup path, backup is disabled");
-                        continue;
-                    }
-
-                    info!("backup db to {}", backup_path);
-                    if let Err(e) = engine.backup_at(backup_path) {
-                        error!("fail to backup: {}", e);
-                    }
-                    info!("backup done");
-                }
+                SIGUSR2 => profile_memory(&profiling_memory),
                 // TODO: handle more signal
                 _ => unreachable!(),
             }
         }
     }
+
+    fn profile_memory(flag: &Arc<AtomicBool>) {
```

Comment:
I think we should move this to the util mod and add a simple test. 

Later we can profile through HTTP API, not only signal.